spec/function.dd

```diff
@@ -115,8 +116,14 @@ $(GNAME MissingFunctionBody):
     $(D ;)
     $(GLINK FunctionContracts)$(OPT) $(GLINK InOutContractExpression) $(D ;)
     $(GLINK FunctionContracts)$(OPT) $(GLINK InOutStatement)
+
+$(GNAME ShortenedFunctionBody):
+    $(D =>) $(GLINK AssignExpression) $(D ;)
```

Comment:
The parser accepts the fully Expression rather than AssignExpression, so one of them is wrong/(needs changing). 

This could also use an example as well (just a cleaned up testcase would be fine)

Reply:
Added an example.

Re: Expression vs AssignExpression, AssignExpression is what's used in the grammar for lambdas, which this is supposed to match. And the only other thing included in Expression is CommaExpression, which would be rejected by semantic analysis anyway. So I think it's the parser that's wrong here.